src/adaptive/index.es

```diff
@@ -1,36 +1,21 @@
-import React, { PureComponent } from 'react';
+import React, { useEffect, useState } from 'react';
 import eventproxy from 'nebenan-helpers/lib/eventproxy';
 import { getMedia, media } from 'nebenan-helpers/lib/dom';
 import { getDisplayName, getForwardedComponent } from '../utils';
 
 
-const makeAdaptive = (Component, options = {}) => {
-  class AdaptiveComponent extends PureComponent {
-    constructor(props) {
-      super(props);
-      this.state = { isMobile: null };
-      this.detectDevice = this.detectDevice.bind(this);
-    }
-
-    componentDidMount() {
-      this.detectDevice();
-      this.stopListeningToResize = eventproxy('resize', this.detectDevice);
-    }
-
-    componentWillUnmount() {
-      this.stopListeningToResize();
-    }
+export const useIsMobile = () => {
+  const [isMobile, setIsMobile] = useState(null);
+  useEffect(() => eventproxy('resize', () => setIsMobile(!getMedia(window, media.mediaM))), []);
```

Comment:
unsubscribe

Reply:
callback passed to useEffect can return a function will be called on unmounted